jmclient/jmclient/bond_calc.py

```diff
@@ -0,0 +1,78 @@
+"""
+Utilities to calculate fidelity bonds values and statistics.
+"""
+from bisect import bisect_left
+from datetime import datetime
+from statistics import quantiles
+from typing import Optional, Dict, Any, Mapping, Tuple, List
+
+from dateutil.relativedelta import relativedelta
+from jmclient import FidelityBondMixin, jm_single, get_interest_rate
+
+
+def add_one_month(dt: datetime) -> datetime:
+    return dt + relativedelta(months=1)
+
+
+def get_bond_values(amount: int,
+                    months: int,
+                    confirm_time: Optional[float] = None,
+                    interest: Optional[float] = None,
+                    exponent: Optional[float] = None,
+                    orderbook: Mapping[str, Any] = None) -> Tuple[Dict[str, Any], List[Dict[str, Any]]]:
+    """
+    Conveniently generate values [and statistics] for multiple possible fidelity bonds.
+
+    Args:
+        amount: Fidelity bond UTXO amount in satoshi
+        months: For how many months to calculate the results
+        confirm_time: Fidelity bond UTXO confirmation time as timestamp, if None, current time is used.
+                      I.e., like if the fidelity bond UTXO with given amount has just confirmed on the blockchain.
+        interest: Interest rate, if None, value is taken from config
+        exponent: Exponent, if None, value is taken from config
+        orderbook: Orderbook data, if given, additional statistics are included in the results.
+    Returns:
+        A tuple with 2 elements.
+         First is a dictionary with all the parameters used to perform fidelity bond calculations.
+         Second is a list of dictionaries, one for each month, with the results.
+    """
+    current_time = datetime.now().timestamp()
+    if confirm_time is None:
+        confirm_time = current_time
+    if interest is None:
+        interest = get_interest_rate()
+    if exponent is None:
+        exponent = float(jm_single().config.get("POLICY", "bond_value_exponent"))
+    else:
+        jm_single().config.set("POLICY", "bond_value_exponent", str(exponent))
```

Comment:
Our difficulties with the config not being dynamically updated come back to bite us again here. If you call this function during long running process (as is the case with `jmwalletd.py`, the daemon we use for RPC-API access) then this updated value will persist, presumably unintentionally. So if doing it this way, you'll have to reset at the end of this function (or wherever).

Reply:
Oh yeah, missed that.

6f68498dbf38fa2be4a2f5b298b49a548dfbe877 reset the value, if modified.
Given the async nature of JM, can this cause problems anyway if the caller is doing other stuff in parallel? As of right now the function is very simple and fast, so shouldn't be a problem for the most part, but is it a theoretical concern?